lib/util/Ordering.ts

```diff
@@ -22,8 +22,21 @@ export function orderTypes(litA: RDF.Term | undefined, litB: RDF.Term | undefine
 
 // Effective ordering
 export function order(orderA: T.Literal<any>, orderB: T.Literal<any>, isAscending: boolean): -1 | 0 | 1 {
+  // If the two values have not the same typedValue type and, hence, are not comparable,
+  // we compare them using the typedValue type name.
+  const aType = getValueClassName(orderA.typedValue);
```

Comment:
Not sure if we'll catch all of the type distinctions by looking at the class name here.
For example, langstrings and plain strings will both produce the `string` type here.

It might be safer to instead check the `dataType` field of the literal.

Reply:
Thank you for your review!

Yes, indeed, the current code leads to orders like "a"@en < "b" < "c"@en.

To implement `dataType` lookup, I would need to change the `order` function signature to get the `typeDiscoveryCallback` value in order to make sure to properly compare comparable data types like `xsd:integer` and `xsd:decimal`. This method does not seem to be used anywhere, is it fine if I make it private to be able to tweak it easily?

Reply:
Indeed, that function is not used anywhere (AFAICS), so it's fine to make it private.

I'm not sure if invoking `typeDiscoveryCallback` will be needed though, because some preprocessing (using `typeDiscoveryCallback`) on the terms is being done here: https://github.com/comunica/sparqlee/blob/a556549f7d366ae67b1e7f2a21e8bc7799e3d9b3/lib/util/Ordering.ts#L16-L17
But I don't know the details of this logic, so don't take my word for it 🙂

Reply:
Thank you! Indeed, there might be something to be done by using the `T.Literal` subclasses ids. The `dataType` field in the `T.Literal` is the original datatype and not the resolved one. I'm going to try something and update the PR for review.

Reply:
I have updated the PR to sort first using explicit "buckets" based on the typed term class names.